### ISSUE TITLE
omnifunc completion for built-in functions

### ISSUE THREAD

**Issue by drichardson** [drichardson/vim-vex#2]
`vcc -X surface` provides information about each function, including return type and parameter types. Use this to implement omnifunc completion for built-in functions.

This feature does not include support for user define function auto-completion.

**Comment by drichardson** [comment]
The above command `vcc -X surface` dumps functions available in the surface context. [There are other contexts as well](https://www.sidefx.com/docs/houdini/vex/contexts/index.html):

```
$ vcc -X
surface
displace
light
shadow
fog
chop
sop
cop2
image3d
cvex
```

Houdini creates a temporary file for the external editor to work with. I'm not sure if the context is available, so I might have to include all the contexts.

**Comment by drichardson** [comment]
Some ideas from the [python completer that ships with Vim include](https://github.com/vim-scripts/pythoncomplete/blob/master/ftplugin/pythoncomplete.vim):

- Show function documentation in scratch buffer on completion
- Show number of arguments in omni complete, though when completion actually occurs only the function name and open parenthesis are written

**Comment by drichardson** [comment]
Implemented in c293d1ee91bb003985a0e0d9d15d38e06b1bf9be.
